src/System.Binary.Base64/System/Binary/Base64.cs

```diff
@@ -135,25 +142,37 @@ public static int Encode(ReadOnlySpan<byte> source, Span<byte> destination)
             while (sourceIndex < srcLength - 2)
             {
                 result = Encode(ref Unsafe.Add(ref srcBytes, sourceIndex));
+                if (destIndex > destLength - 4) goto FalseExit;
```

Comment:
For reference, here is the diff of the disassembly:
![image](https://cloud.githubusercontent.com/assets/6527137/26388322/83af5f82-4008-11e7-8fe3-bcfa7e6f0b30.png)